### ISSUE TITLE
Can't get it working

### ISSUE THREAD

**Issue by throughnothing** [adience-code/logstash-output-kinesis#1]
Hey, I'm really interested in using this plugin, but I can't quite get it running.  Can you provide some better documentation on how to install this plugin into a running logstash instance?  I'm also happy to help debug any issues or improve the code once I can get it working.

**Comment by throughnothing** [comment]
Also, I realize there are changes coming to logstash-1.5, I am trying to get this working with 1.4.2 and everything I can find about installing third party plugins relates to 1.5, which seems to make it much nicer and easier, but nonetheless I'm currently stuck on 1.4.2 and would love to get it working there if possible.  Thanks.

**Comment by navatm** [comment]
Hey William,

I've developed the plugin on logstash 1.5 beta and currently trying to port
it to 1.5 rc4. Could you send me the errors that you are getting? I might
be able to help.

Cheers

**Comment by throughnothing** [comment]
I'm just trying to build the plugin, and get:

```
$ bundle install
Could not find gem 'logstash-devutils (>= 0) ruby' in the gems available on this machine.
```

It seems it doesn't know to look for the java version, and is looking for the ruby version

**Comment by navatm** [comment]
Same error for me when trying to port to 1.5 rc4.
I'll keep you updated once I get it solved.
Please tell me if you find out how to bundle update, since I've used
aws-sdk 2.0.

**Comment by throughnothing** [comment]
I'm not sure if this will be helpful or not, but it seems relevant:

https://github.com/bundler/bundler/issues/2524

**Comment by navatm** [comment]
Should be possible since aws-sdk-v1 can be used side by side with the
previous aws-sdk 2.

**Comment by throughnothing** [comment]
If i do `gem build ...` it "works" though... even though `bundle install failed`...

```
[throughnothing@~/Projects/logstash-output-kinesis](master)
[WMBP] : gem build logstash-output-kinesis.gemspec
WARNING:  open-ended dependency on logstash-mixin-aws (>= 0) is not recommended
  if logstash-mixin-aws is semantically versioned, use:
    add_runtime_dependency 'logstash-mixin-aws', '~> 0'
WARNING:  open-ended dependency on stud (>= 0) is not recommended
  if stud is semantically versioned, use:
    add_runtime_dependency 'stud', '~> 0'
WARNING:  open-ended dependency on logstash-devutils (>= 0, development) is not recommended
  if logstash-devutils is semantically versioned, use:
    add_development_dependency 'logstash-devutils', '~> 0'
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: logstash-output-kinesis
  Version: 0.1.3
  File: logstash-output-kinesis-0.1.3.gem
```

**Comment by throughnothing** [comment]
Now we're getting somewhere:

https://github.com/elastic/logstash/issues/2595

**Comment by throughnothing** [comment]
ok, I think I figured it out, but I get this:

```
Bundler could not find compatible versions for gem "aws-sdk":
 In Gemfile:
   logstash-output-kinesis (>= 0) java depends on
     aws-sdk (~> 2.0) java

   logstash-mixin-aws (>= 0) java depends on
     aws-sdk (~> 1.61.0) java
```

Do you think its ok to remove the aws-sdk dependency, since logstash-mixin-aws requires it, or at least bump the version down to the same major version that it requires (1 vs 2)?

**Comment by throughnothing** [comment]
I'm building it that way for now, by dropping the dep of `aws-sdk` here to `1.61.0` and will see what happens.

**Comment by throughnothing** [comment]
You can see what i've done here to get it building.  I'm going to try installing it into a logstash-1.5.rc4 instance and see what happens...

https://github.com/adience-code/logstash-output-kinesis/compare/master...throughnothing:1.5-fixes

**Comment by throughnothing** [comment]
Ok, I tested in Logstash 1.5, and found some more issues that I've resolved using this post from Amazon:

http://ruby.awsblog.com/post/TxFKSK2QJE6RPZ/Upcoming-Stable-Release-of-AWS-SDK-for-Ruby-Version-2

It seems aws-sdk-v1 and v2 can play nice, so I've updated my branch to reflect that.

https://github.com/adience-code/logstash-output-kinesis/compare/master...throughnothing:1.5-fixes

That ^^ now works for me on logstash 1.5, though I think the `event_partition_key` field should maybe have better defaults, as `message` will likely be too big (> 256 bytes), and `@uuid` seems to be nonstandard (I don't have it), so for now I'm testing with `@timestamp` though I don't know if that's the best.

Would love to hear your thoughts on what I'm doing.
